tests/cursor/bug1839-002.phpt

```diff
@@ -0,0 +1,46 @@
+--TEST--
+PHPC-1839: Referenced local non-interned string in typeMap
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php skip_if_not_live(); ?>
+<?php skip_if_not_clean(); ?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+// Assemble the string so as to not have an interned string
+$rootValue = chr(ord('a')) . 'rray';
+$documentValue = chr(ord('a')) . 'rray';
+
+$typemap = ['root' => &$rootValue, 'document' => &$documentValue];
+
+$manager = new MongoDB\Driver\Manager(URI);
+$cursor = $manager->executeQuery(NS, new MongoDB\Driver\Query([]));
+
+echo "Before:\n";
+debug_zval_dump($typemap);
+
+$cursor->setTypemap($typemap);
+
+echo "After:\n";
+debug_zval_dump($typemap);
+
+?>
+===DONE===
+<?php exit(0); ?>
+--EXPECT--
+Before:
+array(2) refcount(2){
+  ["root"]=>
+  &string(5) "array" refcount(1)
```

Comment:
How does `debug_zval_dump()` display a reference with `refcount(1)` in light of https://github.com/php/php-src/commit/2eb980f614ab59aa8f632ae5a70f719f34e8410f. Wouldn't the `&` be suppressed?

Reply:
Honestly, I have no clue why PHP would show the correct values here. Maybe @sgolemon can provide some insight?